sigmastate/src/main/scala/sigmastate/UnprovenTree.scala

```diff
@@ -35,56 +33,82 @@ trait ProofTreeConjecture extends ProofTree {
   val children: Seq[ProofTree]
 }
 
-
+/**
+  * A node of a sigma-tree used by the prover
+  */
 sealed trait UnprovenTree extends ProofTree {
+
+  /**
+    * Positon of the node in the tree, see comments for `position` field in
+    * `sigmastate.interpreter.Hint`
+    */
+  val position: String
```

Comment:
Could some ADT or newtype be a better fit then String type here?

Reply:
Made NodePosition ADT